### PR TITLE
Rspec test

### PR DESCRIPTION
**Pull request by ioilmio**
We added test for Board and player class methods including:
---------------------
 For Player class: 
-unique?
-player_array
-player_totem
-win?
-tie?
-----------------
For Board class:
-valid?
-who_player

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require './lib/board'
+require './lib/player'
```
Comment (suretrust) [inline spec/board_spec.rb]: This should be `require_relative`. `require` should be used for external files, like gems, while `require_relative` should be used for referring to files within your directory.

File: spec/player_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+require './lib/board'
+require './lib/player'
```
Comment (suretrust) [inline spec/player_spec.rb]: This should be require_relative. require should be used for external files, like gems, while require_relative should be used for referring to files within your directory.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 However, some vitals are missing. 👇 

- [x] Add linter to your repository.

- [x] Let's update your readme to look more professional. Don't forget to add how to run the test.

**Kindly submit for another code review once these issues have been resolved.** _Happy coding!_ 🎉

File: spec/player_spec.rb (hunk 2 above)
Comment (ioilmio) [reply spec/player_spec.rb]: fixed

File: spec/board_spec.rb (hunk 1 above)
Comment (ioilmio) [reply spec/board_spec.rb]: fixed

**Comment by ioilmio** [comment]
The readme get updated and the linter should be in place

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @ioilmio, @guirecordon, good work making the requested changes:+1:, but your test is now broken**

:point_right:  When you use require_relative, you should specify a relative part to the resource.
In your case, the path to your resource should start with two dots, referring to the parent folder before the  lib folder. For example `require_relative './lib/board'` should be `require_relative '../lib/board'`.

:point_right: **[Optional]** Also, your tests should cover both positive and negative scenarios. Sorry, this wasn't pointed out before now. For example, your tie and win methods should be tested for when they are expected to return false, all win conditions should be tested. 

_**Kindly make this change and verify that your tests are passing before submitting another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by guirecordon** [comment]
Thanks for this review, Moses. We fixed the relative paths and added negative test cases.

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀
